contrib/sarplus/python/pysarplus/SARPlus.py

```diff
@@ -428,3 +434,27 @@ def recommend_k_items_slow(self, test, top_k=10, remove_seen=True):
         )
 
         return self.spark.sql(query)
+
+    def recommend_k_items(
+        self,
+        test,
+        top_k=10,
+        remove_seen=True,
+        use_cache=False,
+        n_user_prediction_partitions=200,
+    ):
+        """Recommend top K items for all users which are in the test set.
+
+        Args:
+            test: test Spark dataframe
+            top_k: top n items to return
+            remove_seen: remove items test users have already seen in the past from the recommended set
+            use_cache: use specified local directory stored in self.cache_path as cache for C++ based fast predictions
+            n_user_prediction_partitions: prediction partitions
+        """
+        if not use_cache:
+            return self._recommend_k_items_slow(test, top_k, remove_seen)
+        elif self.cache_path is None:
```

Comment:
I think it would be slightly clearer if this `if` statement was moved under the `else` clause in line 459.

Reply:
Done